File: LeChat.Messaging/LeChat.Messaging.Domain/Entities/Message.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace LeChat.Messaging.Domain.Entities
{
    public class Message
    {
        [Key]
        [Required]
        public Guid Id { get; set; }

        [ForeignKey("Sender")]
        public Guid SenderId { get; set; }

        [ForeignKey("Recipient")]
        public Guid RecipientId { get; set; }

        [ForeignKey("ChatSession")]
        public Guid SessionId { get; set; }

        public string Text { get; set; }
        public DateTime SentTimestamp { get; set; }
        public DateTime? SeenTimestamp { get; set; }

        public virtual User Sender { get; set; }
        public virtual User Recipient { get; set; }
        public virtual ChatSession ChatSession { get; set; }
    }
}

File: LeChat.Client/ClientUI/src/app/shared/models/API/Message.ts
import { User } from "../common/User";
export interface Message {
    id: string;
    senderId: string;
    recipientId: string;
    sessionId: string;
    text: string;
    sentTimestamp: Date;
    seenTimestamp?: Date;

    sender: User;
    recipient: User;
}

